## 1. Import some Libraries, Read Dataset and Exploratory Dataset using Pandas

In [125]:
"""
Meng-import beberapa library dari bahasa pemrograman python yang bertujuan
untuk eksplorasi, membersihkan/mentransformasikan data serta pembuatan grafik visualisasi untuk di analisis
"""
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
plt.style.use("fast")
from sklearn.preprocessing import LabelEncoder

In [126]:
# Melakukan proses pembacaan dataset dari sub-folder "Dataset" serta menampilkan 17 data teratas
data_patient_treatment = pd.read_csv("../Dataset/patient_treatment_classification_dataset.csv")

data_patient_treatment.head(17)

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
0          35.1          11.8         4.65        6.3          310  25.4   
1          43.5          14.8         5.39       12.7          334  27.5   
2          33.5          11.3         4.74       13.2          305  23.8   
3          39.1          13.7         4.98       10.5          366  27.5   
4          30.9           9.9         4.23       22.1          333  23.4   
5          34.3          11.6         4.53        6.6          185  25.6   
6          31.1           8.7         5.06       11.1          416  17.2   
7          40.3          13.3         4.73        8.1          257  28.1   
8          33.6          11.5         4.54       11.4          262  25.3   
9          35.4          11.4         4.80        2.6          183  23.8   
10         33.7          11.5         4.57       13.2          322  25.2   
11         54.0          16.6         7.61       10.0           88  21.8   
12         31.7          10.4         4.91        9.7          348  21.2   
13         35.3          11.9         4.40        5.8          205  27.0   
14         34.5           9.8         5.75       15.4          548  17.0   
15         34.0          10.3         5.27       16.2          572  19.5   
16         35.0          11.6         4.58        7.4          154  25.3   

    MCHC   MCV  AGE SEX SOURCE  
0   33.6  75.5    1   F    out  
1   34.0  80.7    1   F    out  
2   33.7  70.7    1   F    out  
3   35.0  78.5    1   F    out  
4   32.0  73.0    1   M    out  
5   33.8  75.7    1   M    out  
6   28.0  61.5    1   F    out  
7   33.0  85.2    1   F    out  
8   34.2  74.0    1   F    out  
9   32.2  73.8    1   F    out  
10  34.1  73.7    1   M    out  
11  30.7  71.0    1   F     in  
12  32.8  64.6    1   M     in  
13  33.7  80.2    1   M    out  
14  28.4  60.0    1   M    out  
15  30.3  64.5    1   M    out  
16  33.1  76.4    1   F    out

Berikut adalah informasi atas penjelasan lengkap mengenai 11 kolom data yang ada di dalam variabel DataFrame yang bernama "data_patient_treatment" diantaranya adalah sebagai berikut:

1. HAEMATOCRIT: Kolom yang berisikan data terkait hasil uji laboratorium mengenai hematokrit pasien. Kolom data ini menunjukkan bagian dari pemeriksaan darah untuk mendeteksi apakah pasien tersebut mengalami anemia/kekurangan sel darah merah atau tidak.
2. HAEMOGLOBINS: Kolom yang berisikan data terkait hasil uji laboratorium mengenai hemoglobin atau sel darah merah milik masing-masing pasien.
3. ERYTHROCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai eritrosit atau sel darah putih milik masing-masing pasien.
4. LEUCOCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai leukosit milik masing-masing pasien.
5. THROMBOCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai trombosit pada masing-masing pasien.
6. MCH: Kolom yang berisikan data terkait hasil uji laboratorium terhadap masing-masing pasien mengenai MCH atau Mean Corpuscular Hemoglobin). MCH merupakan jumlah rata-rata hemoglobin di dalam sel darah merah milik pasien.
7. MCHC: Kolom yang berisikan data terkait hasil uji lab pasien mengenai MCHC atau Mean Corpuscular Hemoglobin Concentration. MCHC merupakan perhitungan rata-rata terhadap konsentrasi hemoglobin di dalam eritrosit pasien.
8. MCV: Kolom yang berisikan data terkait hasil uji lab pasien mengenai MCV atau Mean Corpuscular Volume. MCV merupakan ukuran rata-rata dari sel darah merah milik pasien.
9. AGE: Kolom yang berisikan data mengenai umur dari setiap pasien.
10. SEX: Kolom yang berisikan data mengenai jenis kelamin pasien yang ditandai dengan nilai F (female/perempuan) dan M (male/laki-laki).
11. SOURCE: Terakhir adalah kolom yang berisikan data dari masing-masing status pasien apakah pasien tersebut merupakan pasien rawat inap (ditandai dengan data "in") atau pasien rawat jalan (ditulis/ditandai dengan data "out"). Kolom ini juga merupakan kolom target yang akan dijadikan sebagai label klasifikasi pada tahap pemodelan machine learning.

Selain informasi atas 11 kolom yang ada di dalam "data_patient_treatment" yang sudah dijelaskan di atas, kalian dapat mengunjungi sumber dataset dari Kaggle melalui link di bawah ini:

https://www.kaggle.com/datasets/saurabhshahane/patient-treatment-classification

Eksplorasi selanjutnya setelah menampilkan 17 data teratas dari "data_patient_treatment" adalah membuktikan apakah variabel "data_patient_treatment" merupakan sebuah DataFrame atau Series dengan cara mengetahui dimensi dari "data_patient_treatment". Jika dimensinya adalah 1 maka variabel "data_patient_treatment" merepresentasikan sebagai Series sedangkan jika dimensinya adalah 2 maka merepresentasikan sebagai DataFrame.

In [127]:
"""
Menampilkan nilai dimensi yang dimiliki oleh variabel "data_patient_treatment"
untuk mengetahui apakah variabel tersebut merupakan Series/DataFrame
"""
print("Nilai Dimensi dari Variabel 'data_patient_treatment' adalah (1: Series | 2: DataFrame): ", data_patient_treatment.ndim)

Nilai Dimensi dari Variabel 'data_patient_treatment' adalah (1: Series | 2: DataFrame):  2


In [128]:
# Menampilkan 5 data terbawah yang ada di dalam variabel DataFrame "data_patient_treatment"
data_patient_treatment.tail()

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
4407         32.8          10.4         3.49        8.1           72  29.8   
4408         33.7          10.8         3.67        6.7           70  29.4   
4409         33.2          11.2         3.47        7.2          235  32.3   
4410         31.5          10.4         3.15        9.1          187  33.0   
4411         33.5          10.9         3.44        5.8          275  31.7   

      MCHC    MCV  AGE SEX SOURCE  
4407  31.7   94.0   92   F     in  
4408  32.0   91.8   92   F     in  
4409  33.7   95.7   93   F    out  
4410  33.0  100.0   98   F     in  
4411  32.5   97.4   99   F    out

In [129]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel "data_patient_treatment" secara terpisah
print("1. Menampilkan Jumlah Baris dan Kolom Data secara Terpisah: ")
print("Jumlah Baris Data yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment': ", len(data_patient_treatment))

print("Jumlah Kolom yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment': ",
      len(data_patient_treatment.columns), "\n")

# Menampilkan jumlah baris dan kolom dari variabel "data_patient_treatment" tanpa terpisah
print("2. Menampilkan Jumlah Baris dan Kolom tanpa Terpisah: ")
print("Jumlah Baris dan Kolom Data dari Variabel DataFrame 'data_patient_treatment': ", data_patient_treatment.shape)

1. Menampilkan Jumlah Baris dan Kolom Data secara Terpisah: 
Jumlah Baris Data yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment':  4412
Jumlah Kolom yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment':  11 

2. Menampilkan Jumlah Baris dan Kolom tanpa Terpisah: 
Jumlah Baris dan Kolom Data dari Variabel DataFrame 'data_patient_treatment':  (4412, 11)


In [130]:
# Menampilkan nama kolom dari variabel "data_patient_treatment"
data_patient_treatment.columns

Index(['HAEMATOCRIT', 'HAEMOGLOBINS', 'ERYTHROCYTE', 'LEUCOCYTE',
       'THROMBOCYTE', 'MCH', 'MCHC', 'MCV', 'AGE', 'SEX', 'SOURCE'],
      dtype='object')

Berdasarkan hasil diatas dapat diketahui jika variabel DataFrame "data_patient_treatment" memiliki 4.412 data yang tersebar ke dalam 11 kolom berbeda. Kesebelas kolom tersebut adalah HAEMATOCRIT, HAEMOGLOBINS, ERYTHROCYTE, LEUCOCYTE, THROMBOCYTE, MCH, MCHC, MCV, AGE, SEX dan SOURCE.

Namun jika diperhatikan penamaan kolom tersebut masih belum tepat serta ditulis menggunakan huruf kapital sehingga harus di-update menjadi format penamaan kolom yang sesuai. Proses transformasi terhadap penamaan kolom akan dilakukan pada bagian selanjutnya tepatnya pada bagian Data Preprocessing.

In [131]:
# Menampilkan tipe data apa saja yang dimiliki oleh variabel DataFrame "data_patient_treatment"
data_patient_treatment.dtypes

HAEMATOCRIT     float64
HAEMOGLOBINS    float64
ERYTHROCYTE     float64
LEUCOCYTE       float64
THROMBOCYTE       int64
MCH             float64
MCHC            float64
MCV             float64
AGE               int64
SEX              object
SOURCE           object
dtype: object

Berdasarkan hasil eksplorasi diatas menggunakan fungsi dtypes dari pandas, dapat diketahui bahwa setidaknya ada 3 tipe data yang berbeda yaitu float, int dan object. Tipe data float merupakan tipe data yang menunjukkan/identitas dari data berupa bilangan numerik pecahan. Tipe data int atau integer merupakan tipe data yang menjadi indetitas dari data numerik bilangan bulat. Sedangkan tipe data object merupakan tipe data yang menunjukkan identitas dari data kategorik atau data string seperti huruf, simbol karakter atau gabungan antara bilangan numerik (pecahan/bulat) dengan huruf/angka. 

In [132]:
# Menampilkan jumlah total dari elemen yang dimiliki oleh variabel DataFrame
data_patient_treatment.size

48532

Kenapa jumlah total element dari "data_patient_treatment" bernilai 48.532? Hal tersebut dikarenakan jumlah total element yang ada dimiliki oleh fungsi "size" merupakan sebuah bentuk bilangan hasil kalkulasi perkalian antara jumlah baris (number of rows) dengan jumlah kolom (number of columns) yang dimiliki oleh variabel DataFrame (dalam hal ini adalah data_patient_treatment).

Berarti jumlah baris dari "data_patient_treatment" yang bernilai 4.412 * jumlah kolom yang saat ini dimiliki oleh "data_patient_treatment" yaitu 11 sehingga menghasilkan 48.532 element. Untuk lebih jelasnya dapat lihat perhitungan yang ada di bawah ini:

4412 * 11 = 48532

In [133]:
# Menampilkan jumlah total data yang terindikasi sebagai data duplikat
data_patient_treatment.duplicated().sum()

0

Hasil diatas menunjukkan bahwa data yang dimiliki oleh "data_patient_treatment" tidak ada yang terduplikasi. Data duplikat merupakan data yang memiliki nilai yang sama di setiap kolomnya serta pada baris yang lain juga terdapat nilai data yang serupa. Selanjutnya saya akan mengecek apakah di dalam "data_patient_treatment" terdapat missing value di masing-masing kolom atau tidak dengan menggunakan fungsi "isnull()" dan "sum()".

In [134]:
"""
Menampilkan jumlah keseluruhan missing value yang dimiliki oleh
masing-masing kolom dari variabel DataFrame "data_patient_treatment"
"""
data_patient_treatment.isnull().sum()

HAEMATOCRIT     0
HAEMOGLOBINS    0
ERYTHROCYTE     0
LEUCOCYTE       0
THROMBOCYTE     0
MCH             0
MCHC            0
MCV             0
AGE             0
SEX             0
SOURCE          0
dtype: int64

Menggunakan fungsi "isnull()" dan "sum()" secara bersamaan bertujuan untuk menghitung jumlah total keseluruhan dari data yang hilang (missing value) dari masing-masing kolom yang ada di dalam variabel DataFrame "data_patient_treatment". Dengan menggabungkan kedua fungsi tersebut dapat secara cepat dan tepat mengetahui berapa banyak missing value/data yang hilang dari masing-masing kolom yang berada di dalam variabel DataFrame "data_patient_treatment".

In [135]:
"""
Menampilkan ringkasan informasi singkat yang dimiliki oleh DataFrame, baik informasi per masing-masing
kolom seperti missing value, tipe data, dll, informasi jumlah kolom & baris dan penggunaan memori
"""
data_patient_treatment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4412 entries, 0 to 4411
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   HAEMATOCRIT   4412 non-null   float64
 1   HAEMOGLOBINS  4412 non-null   float64
 2   ERYTHROCYTE   4412 non-null   float64
 3   LEUCOCYTE     4412 non-null   float64
 4   THROMBOCYTE   4412 non-null   int64  
 5   MCH           4412 non-null   float64
 6   MCHC          4412 non-null   float64
 7   MCV           4412 non-null   float64
 8   AGE           4412 non-null   int64  
 9   SEX           4412 non-null   object 
 10  SOURCE        4412 non-null   object 
dtypes: float64(7), int64(2), object(2)
memory usage: 379.3+ KB


Berdasarkan output yang didapatkan dari fungsi "info()", kita dapat mengetahui beberapa informasi yang penting terkait variabel DataFrame "data_patient_treatment" diantaranya adalah sebagai berikut:

1. Variabel "data_patient_treatment" merupakan sebuah DataFrame yang berisikan sebuah dataset tentang hasil pemeriksaan laboratorium untuk pasien yang digunakan untuk menentukan penanganan pasien selanjutnya apakah pasien tersebut harus di rawat inap atau rawat jalan.
2. Variabel DataFrame "data_patient_treatment" memiliki jumlah data/baris sebanyak 4.412 data. Selain itu, variabel "data_patient_treatment" memiliki 11 kolom yang diantaranya adalah "HAEMATOCRIT", "HAEMOGLOBINS", "ERYTHROCTYE", "LEUCOCYTE", "THROMBOCYTE", "MCH", "MCHC", "MCV", "AGE", "SEX" dan "SOURCE".
3. Setiap kolom yang ada di dalam "data_patient_treatment" tidak memiliki nilai NaN atau nilai kosong di dalam kolom-kolom data. Itu berarti kesebelas kolom yang ada di dalam "data_patient_treatment" tidak memiliki missing value di dalamnya.
4. Terdapat 3 variasi tipe data yang ada di dalam variabel DataFrame "data_patient_treatment" yaitu tipe data float untuk data angka pecahan. Tipe data int (integer) untuk data angka bilangan bulat dan tipe data object (string) yang merupakan data karakter & huruf.
5. Tipe data yang paling banyak digunakan oleh kolom-kolom dari "data_patient_treatment" adalah float sedangkan tipe data int atau object (string) hanya digunakan oleh masing-masing 2 kolom saja. Teruntuk kolom yang menggunakan tipe data float adalah "HAEMATOCRIT", "HAEMOGLOBINS", "ERYHROCYTE", "LEUCOCYTE", "MCH", "MCHC" dan "MCV". Lalu kolom yang menggunakan tipe data int adalah "THROMBOCYTE" dan "AGE". Terakhir adalah kolom "SEX" dan "SOURCE" yang menggunakan tipe data object (string).

In [136]:
"""
Menampilkan nilai statistik deskriptif untuk kolom bersifat angka
numerik (float & int) yang ada di dalam "data_patient_treatment"
"""
data_patient_treatment.describe()

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE    LEUCOCYTE  THROMBOCYTE  \
count  4412.000000   4412.000000  4412.000000  4412.000000  4412.000000   
mean     38.197688     12.741727     4.541260     8.718608   257.524479   
std       5.974784      2.079903     0.784091     5.049041   113.972365   
min      13.700000      3.800000     1.480000     1.100000     8.000000   
25%      34.375000     11.400000     4.040000     5.675000   188.000000   
50%      38.600000     12.900000     4.570000     7.600000   256.000000   
75%      42.500000     14.200000     5.050000    10.300000   321.000000   
max      69.000000     18.900000     7.860000    76.600000  1183.000000   

               MCH         MCHC          MCV          AGE  
count  4412.000000  4412.000000  4412.000000  4412.000000  
mean     28.234701    33.343042    84.612942    46.626473  
std       2.672639     1.228664     6.859101    21.731218  
min      14.900000    26.000000    54.000000     1.000000  
25%      27.200000    32.700000    81.500000    29.000000  
50%      28.700000    33.400000    85.400000    47.000000  
75%      29.800000    34.100000    88.700000    64.000000  
max      40.800000    39.000000   115.600000    99.000000

Berdasarkan hasil diatas terkait menampilkan nilai statistik untuk kolom yang bersifat/memiliki data angka, dapat diketahui bahwa nilai-nilai statistik dari kolom seperti "HAEMATOCRIT", "HAEMOGLOBINS", "ERYTHROCYTE", "LEUCOCYTE", "THROMBOCYTE", "MCH", "MCHC", MCV" dan kolom "AGE" adalah nilai count, mean/average, std/standar deviasi, min, 25% atau quartil 1, 50% atau quartil 2, 75% atau quartil 3 dan max. Delapan nilai statistik deskriptif tersebut secara default diperuntukkan untuk kolom-kolom yang memiliki data angka baik angka pecahan/bilangan bulat.

Sedangkan kolom seperti "SEX" dan "SOURCE" tidak memiliki delapan nilai statistik tersebut dikarenakan kedua kolom tersebut bersifat kategorikal yang datanya merupakan data huruf dan karakter. Jika ingin melihat nilai statistik deskriptif untuk kolom yang bersifat memiliki tipe data object (string) kita dapat menggunakan kedua cara berikut ini:

1. Cara pertama adalah dengan menuliskan script data_patient_treatment.describe(include="O") jika hanya ingin melihat nilai statistik deskriptif untuk kolom yang memiliki tipe data object.

2. Atau cara kedua adalah adalah dengan data_mentari_niaga.describe(include="all"). Jika menggunakan cara kedua maka seluruh kolom akan menampilkan nilai statistik deskriptifnya masing-masing.

Untuk mengetahui/mengambil kesimpulan secara tepat terkait nilai statistik untuk kolom yang ada di dalam variabel DataFrame "data_patient_treatment", maka dapat dibaca penjelasan di bawah ini:

1. Kolom "HAEMOGLOBINS" memiliki nilai statistik minimum sebesar 4 (3.800000) dan nilai statistik maximum sebesar 19 (18.900000). Selain kedua nilai statistik tersebut (minimum dan maximum), nilai statistik mean/average untuk kolom ini adalah sebesar 13 (12.741727) dengan nilai standar deviasi yaitu 2 (2.079903). Lalu untuk nilai quartil pertama (25%) dari kolom "HAEMOGLOBINS" adalah 11 (11.400000), quartil kedua (50%) atau yang diketahui juga sebagai nilai median menghasilkan nilai 13 (12.900000), dan statistik deskriptif quartil ketiga (75%) memiliki nilai sebesar 14 (14.200000).

2. Kolom kedua yang akan dibahas nilai statistik deskriptifnya adalah "ERYTHROCYTE" dengan nilai statistik yang pertama yaitu maximum yang memiliki nilai sebesar 8 (7.860000) serta nilai statistik minimum yang dimiliki oleh kolom ini adalah 1 (1.480000). Selain kedua nilai statistik tersebut (maximum dan minimum), kolom "ERYTHROCYTE" memiliki nilai mean dan nilai mean yang dimiliki oleh kolom ini adalah 5 (4.541260) dan nilai standar deviasi yaitu 1 (0.784091). Lalu untuk nilai quartil pertama (25%) yang dimiliki oleh kolom ini adalah 4 (4.040000), quartil kedua (50%) atau yang diketahui juga sebagai nilai median memiliki nilai 5 (4.570000), dan statistik deskriptif untuk quartil ketiga (75%) memiliki nilai sebesar 5 (5.050000).

3. Kolom ketiga yang akan dibahas nilai statistik deskriptifnya adalah "MCV" dengan nilai statistik yang pertama yaitu maximum yang memiliki nilai sebesar 116 (115.600000) serta nilai statistik minimum yang dimiliki oleh kolom ini adalah 54 (54.000000). Selain kedua nilai statistik tersebut (maximum dan minimum), kolom "MCV" memiliki nilai mean dan nilai mean yang dimiliki oleh kolom ini adalah 85 (84.612942) dan nilai standar deviasi yaitu 7 (6.859101). Lalu untuk nilai quartil pertama (25%) yang dimiliki oleh kolom ini adalah "MCV" adalah 82 (81.500000), quartil kedua (50%) atau yang diketahui juga sebagai nilai median memiliki nilai 85 (85.400000), dan statistik deskriptif untuk quartil ketiga (75%) memiliki nilai sebesar 89 (88.700000).

4. Kolom keempat yang akan dibahas nilai statistik deskriptifnya adalah "THROMBOCYTE". Nilai statistik pertama yang dimiliki oleh kolom ini adalah nilai rata-rata (mean) yang sebesar 258 (257.524479). Lalu nilai statistik kedua yaitu minimum dengan nilai 8 (8.000000) dan nilai statistik maximum sebesar 1.183 (1183.000000). Selain ketiga nilai statistik tersebut (mean, minimum dan maximum), sebagai kolom yang berisikan data numerik bilangan pecahan juga sudah pasti memiliki nilai statistik yaitu standar deviasi, quartil pertama (25%), quartil kedua/median (50%) dan quartil ketiga (75%). Nilai standar deviasi yang dimiliki oleh kolom "THROMBOCYTE" yaitu 114 (113.972365), nilai quartil pertama (25%) adalah 188 (188.000000), quartil kedua (50%) sebesar 256, dan quartil ketiga (75%) dengan nilai 321 (321.000000).

5. Kolom terakhir yang akan dijadikan contoh bagaimana caranya mengambil kesimpulan/mendapatkan informasi atas interpretasi nilai statistik deskriptif melalui fungsi "describe()" dari library pandas adalah kolom "AGE". Nilai statistik pertama yang dimiliki oleh kolom ini adalah nilai rata-rata (mean) dengan nilai mean sebesar 47 (46.626473). Lalu nilai statistik kedua dan ketiga yang dimiliki yaitu standar deviasi dengan nilai 22 (21.731218) dan nilai statistik minimum yaitu 1 (1.000000). Selain ketiga nilai statistik tersebut (mean, standar deviasi dan minimum), kolom ini juga memiliki nilai statistik quartil dan maximum. Untuk nilai quartil pertama (25%) adalah 29 (29.000000), quartil kedua/median (50%) adalah 47 (47.000000), quartil ketiga (75%) adalah 64 (64.000000) dan nilai statistik deskriptif terakhir yang ada di kolom "AGE" adalah maximum dengan nilai sebesar 99 (99.000000). Ini berarti umur pasien terkecil (paling kecil dan muda) yang dimiliki oleh dataset yang digunakan di dalam penelitian kali ini berumur 1 tahun dan umur pasien tertinggi (paling tinggi/tua) berumur 99 tahun.

In [137]:
# Menampilkan nilai statistik deskriptif untuk kolom data kategorik (kolom "SEX" dan "SOURCE")
data_patient_treatment.describe(include="O")

SEX SOURCE
count   4412   4412
unique     2      2
top        M    out
freq    2290   2628

Seperti yang sudah dijelaskan saat menampilkan nilai statistik deskriptif untuk kolom data numerik, salah satu dari cara untuk menampilkan sekaligus mengetahui nilai statistik deskriptif untuk kolom data kategorik adalah dengan cara "variableDataFrame.describe(include="O")" dimana nilai O adalah Object yang itu berarti menampilkan nilai statistik untuk kolom bertipe data object (string).

Hasil output dari fungsi "describe(include="O")" adalah kedua kolom yang ada di dalam variabel "data_patient"treatment" yaitu kolom "SEX" dan "SOURCE" adalah menampilkan nilai statistik yang dimilikinya. Berbeda ketika saat menampilkan nilai statistik untuk kolom data numerik, nilai statistik untuk kolom data string hanya ada 4 nilai statistik saja yaitu count, unique, top dan freq. Nilai count merupakan nilai yang menghitung ada berapa banyak data yang dimiliki oleh masing-masing kolom. Nilai unique merupakan nilai yang memiliki keunikan tersendiri dan tidak sama antar satu dengan yang lainnya. Sebagai contoh nilai unik yang dapat diketahui adalah jenis kelamin hanya ada 2 yaitu Laki-Laki dan Perempuan. Kedua jenis kelamin tersebut termasuk ke dalam kategori nilai unik. Selanjutnya adalah nilai statistik top yang menunjukkan data yang paling banyak muncul di dalam kolom tersebut. Nilai top dapat diketahui secara jelas dengan menggunakan fungsi "value_counts()". Nilai statistik yang terakhir untuk kolom bertipe data object adalah freq yang merupakan jumlah total kemunculan (frekuensi) data yang paling sering terlihat kemunculannya di kolom tersebut. Untuk lebih jelasnya dapat di lihat penjelasan yang ada di bawah ini:

1. Nilai unique pada kolom "SEX" berjumlah 2 saja yang itu berarti terdapat 2 data yang berbeda pada kolom "SEX". Seperti yang kita tau bahwa jenis kelamin yang ada di dunia hanya ada Laki-Laki dan Perempuan saja. Maka dari itu karena kolom ini menyimpan data jenis kelamin pasien, kolom "SEX" hanya memiliki 2 data unik yang itu berarti jenis kelamin pasien yang tersimpan di dalam Electronic Health Record Predicting yang dikumpulkan dari Rumah Sakit Swasta di Indonesia hanya ada 2 yaitu Laki-Laki dan Perempuan. Selanjutnya dari kedua jenis kelamin tersebut, data yang paling sering terlihat di dalam kolom "SEX" atau dapat dikatakan data yang memiliki jumlah kemunculan terbanyak di kolom "SEX" adalah M atau Male (Laki-Laki). Nilai data M atau Male (Laki-Laki) memiliki jumlah kemunculan (freq) sebanyak 2.290 kali. Jika disimpulkan hal ini berarti kolom "SEX" memiliki data unik yaitu 2 data saja (yaitu Male/Laki-Laki dan Female (F)/Perempuan) dan dari kedua data tersebut, nilai data M/Male yang memiliki jumlah frekuensi kemunculan terbanyak yaitu 2.290 kali.

2. Terakhir adalah kolom "SOURCE" yang memiliki nilai unique hanya 2 saja yang itu berarti terdapat 2 data yang berbeda pada kolom "SOURCE". Selanjutnya dari kedua data unik tersebut, data yang paling sering terlihat di dalam kolom "SOURCE" atau dapat dikatakan data yang memiliki jumlah kemunculan terbanyak di kolom ini adalah out (out care patient). Nilai data out (out care patient) memiliki jumlah kemunculan (freq) terbanyak yaitu berjumlah 2.628 kali. Jika disimpulkan maka kolom "SOURCE" memiliki 2 data unik (yaitu in dan out. Nilai in untuk pasien yang diklasifikasikan sebagai pasien rawat inap dan nilai out untuk pasien yang dikategorikan sebagai pasien rawat jalan). Kedua data tersebut (in dan out) memiliki data yang paling banyak terlihat di dalam kolom "SOURCE" dan data tersebut adalah out. Nilai data out merupakan data yang paling banyak terlihat di dalam kolom "SOURCE" karena data ini (data out) memiliki jumlah frekuensi kemunculan sebanyak 2.628 kali.

In [138]:
# Menampilkan jumlah frekuensi dari masing-masing kedua data unik yang dimiliki oleh kolom "SEX" dan "SOURCE"
print(data_patient_treatment["SEX"].value_counts(), "\n")

print(data_patient_treatment["SOURCE"].value_counts())

M    2290
F    2122
Name: SEX, dtype: int64 

out    2628
in     1784
Name: SOURCE, dtype: int64


Hasil diatas menunjukkan jumlah frekuensi/kemunculan data dari kedua kolom data kategorik yaitu kolom "SEX" dan "SOURCE". Berdasarkan output diatas dapat diketahui bahwa jenis kelamin Laki-Laki/Male (yang ditandai dengan nilai M) memiliki jumlah frekuensi kemunculan data sebanyak 2.290. Jumlah tersebut sudah sangat tepat dengan informasi yang didapatkan ketika menampilkan nilai statistik deskriptif untuk kolom data yang memiliki tipe data object/string. Hal ini berarti menyimpulkan bahwa mayoritas pasien yang melakukan test laboratorium di Rumah Sakit Swasta di Indonesia adalah pasien berjenis kelamin Laki-Laki karena dari 4.412 data yang berhasil tersimpan di dalam Electronic Health Record Predicting, sebanyak 2.290 data mengkategorikan pasien tersebut adalah Laki-Laki sedangkan 2.122 dikategorikan sebagai pasien Perempuan/Female yang di tandai dengan nilai F pada kolom "SEX".

Selain menunjukkan jumlah frekuensi kemunculan data terbanyak dari kolom "SEX" atau jenis kelamin, hasil diatas juga memperlihatkan jumlah kemunculan dua data unik dari kolom "SOURCE" yang diantaranya adalah out dan in. Nilai out merupakan nilai yang menunjukkan status dari seorang pasien rawat jalan sedangkan nilai in merupakan nilai yang menunjukkan status seorang pasien rawat inap. Pasien rawat jalan yang memiliki nilai out di dalam kolom "SOURCE" memiliki jumlah frekuensi kemunculan nilai paling banyak yaitu 2.628 sedangkan pasien rawat inap yang memiliki nilai in hanya memiliki jumlah frekuensi kemunculan nilai yaitu 1.784.

In [139]:
# Menampilkan nilai korelasi antar masing-masing kolom bertipe data int & float dari variabel DataFrame "data_patient_treatment"
data_patient_treatment.corr()

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE  \
HAEMATOCRIT      1.000000      0.973267     0.864989  -0.217218    -0.003562   
HAEMOGLOBINS     0.973267      1.000000     0.818013  -0.208583    -0.045441   
ERYTHROCYTE      0.864989      0.818013     1.000000  -0.186711     0.025879   
LEUCOCYTE       -0.217218     -0.208583    -0.186711   1.000000     0.283262   
THROMBOCYTE     -0.003562     -0.045441     0.025879   0.283262     1.000000   
MCH              0.083714      0.203248    -0.383666  -0.014840    -0.138058   
MCHC             0.108098      0.314159    -0.048313   0.003194    -0.198921   
MCV              0.054948      0.105379    -0.436041  -0.015885    -0.075961   
AGE             -0.254799     -0.239480    -0.412858   0.191987     0.031064   

                   MCH      MCHC       MCV       AGE  
HAEMATOCRIT   0.083714  0.108098  0.054948 -0.254799  
HAEMOGLOBINS  0.203248  0.314159  0.105379 -0.239480  
ERYTHROCYTE  -0.383666 -0.048313 -0.436041 -0.412858  
LEUCOCYTE    -0.014840  0.003194 -0.015885  0.191987  
THROMBOCYTE  -0.138058 -0.198921 -0.075961  0.031064  
MCH           1.000000  0.589830  0.931804  0.350166  
MCHC          0.589830  1.000000  0.259672  0.041741  
MCV           0.931804  0.259672  1.000000  0.395380  
AGE           0.350166  0.041741  0.395380  1.000000

Dengan menggunakan fungsi "corr()" pada variabel "data_patient_treatment" dapat memperlihatkan nilai korelasi/hubungan timbal-balik antar masing-masing kolom tetapi dengan 1 syarat yaitu kolom tersebut harus memiliki data numerik/angka yang itu berarti fungsi "corr()" diperuntukkan untuk kolom bertipe data integer/float.

Agar mendapatkan informasi apakah suatu kolom memiliki hubungan sebab-akibat yang sangat kuat dapat dilihat melalui nilai output yang dihasilkan oleh fungsi "corr()" pada variabel "data_patient_treatment". Berikut adalah beberapa contoh penjelasan terkait nilai korelasi yang dimiliki oleh masing-masing kolom di dalam variabel "data_patient_treatment":

1. Kolom "HAEMATOCRIT" memiliki hubungan sebab-akibat/timbal-balik yang sangat tinggi dengan kolom "HAEMOGLOBINS" dikarenakan nilai korelasi yang dihasilkan oleh kedua kolom tersebut adalah 0.973267. Itu berarti kolom "HAEMATOCRIT" dan "HAEMOGLOBINS" saling memiliki keterkaitan dari data yang dimiliki oleh kedua kolom tersebut. Berbanding terbalik jika kita melihat nilai korelasi antara kolom "HAEMATOCRIT" dengan kolom "THROMBOCYTE" yang hanya -0.003562. Itu berati kedua kolom tersebut tidak memiliki keterkaitan/hubungan timbal-balik yang bagus antar masing-masing data yang dihasilkan.

2. Contoh kedua yang akan dijadikan contoh penjelasan mengenai keterkaitan/hubungan sebab-akibat antar kolom yaitu "HAEMOGLOBINS" dengan "ERYTHROCYTE". Kedua kolom ini memiliki hubungan sebab-akibat yang cukup dekat karena nilai korelasi yang dihasilkan oleh kedua kolom tersebut adalah 0.818013. Sangat berbanding terbalik jika kita melihat hubungan sebab-akibat antara kolom "HAEMOGLOBINS" dengan "LEUCOCYTE", "THROMBOCYTE" ataupun "AGE" yang sama-sama memiliki nilai korelasi minus yaitu -0.208583 (LEUCOCYTE), -0.045441	(THROMBOCYTE) dan -0.239480 (AGE). Itu sudah cukup menjelaskan dan menjadi bukti bahwa kolom "HAEMOGLOBINS" dengan ketiga kolom yaitu "LEUCOCYTE", "THROMBOCYTE" ataupun "AGE" tidak memiliki keterkaitan hubungan antar data.

3. Selanjutnya adalah hubungan timbal-balik yang dimiliki oleh kolom "MCH" dan "MCV" yang memiliki keterkaitan antar data di masing-masing kedua kolom tersebut sangat dekat. Hal tersebut dikarenakan nilai korelasi yang dimiliki oleh kedua kolom ini (kolom "MCH" dah "MCV) sangat tinggi dan hampir mendekati kesempurnaan (sama seperti korelasi antara kolom "HAEMATOCRIT" dan "HAEMOGLOBINS") yaitu 0.931804.

4. Hubungan sebab-akibat dari data yang dimiliki oleh kolom "ERYTHROCYTE" dengan beberapa kolom seperti "LEUCOCYTE", "MCH", "MCHC", "MCV" dan "AGE" sangat rendah karena nilai korelasinya berada di bawah minus. Korelasi/keterkaitan hubungan data antara kolom "ERYTHROCYTE" dengan kolom "LEUCOCYTE" adalah -0.186711, kolom "ERYTHROCYTE" dengan "MCH" adalah -0.383666, dengan kolom "MCHC" hanya -0.048313, dengan kolom "MCV" adalah -0.436041 dan keterkaitan hubungan data dengan kolom "AGE" juga hanya -0.412858. Ini berarti eritrosit yang rendah bukan menjadi sebab utama jika umur pasien terlalu tua/muda. Eritrosit yang rendah juga bukan menjadi penyebab tingginya/rendahnya kandungan leukosit dalam tubuh pasien, dll.

Jadi intinya jika hubungan timbal-balik antar masing-masing kolom sangat tinggi seperti yang ditunjukkan oleh kolom "HAEMATOCRIT" dan "HAEMOGLOBINS" (0.973267) atau "MCH" dengan "MCV" (0.931804) mengindikasikan bahwa data tersebut sangat berkaitan/saling terkait antar satu dengan yang lain.

### 1.1. Melakukan Proses Analisis Sederhana dengan Filter

In [140]:
"""
Menampilkan kolom data berdasarkan tipe data yaitu int dan object dari variabel "data_patient_treatment"
sedangkan tipe data float yang ada di dalam variabel "data_patient_treatment" tidak dimunculkan
"""
data_patient_treatment.select_dtypes(include=["int", "object"]).tail(8)

THROMBOCYTE  AGE SEX SOURCE
4404          110   92   F     in
4405          100   92   F     in
4406          112   92   F     in
4407           72   92   F     in
4408           70   92   F     in
4409          235   93   F    out
4410          187   98   F     in
4411          275   99   F    out

In [141]:
"""
Menampilkan kolom data berdasarkan tipe data float dari "data_patient_treatment"
sedangkan kedua tipe data yang lain (int dan object) tidak dimunculkan
"""
data_patient_treatment.select_dtypes(exclude=["int", "object"]).tail(29)

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE   MCH  MCHC    MCV
4383         30.2           9.8         3.45        6.3  28.4  32.5   87.5
4384         29.8          10.1         3.44        7.8  29.4  33.9   86.6
4385         30.7          10.1         3.48        7.4  29.0  32.9   88.2
4386         28.9           9.6         3.24        8.6  29.6  33.2   89.2
4387         26.3           8.6         2.89        6.3  29.8  32.7   91.0
4388         27.1           8.9         3.01        7.6  29.6  32.8   90.0
4389         25.8           8.3         2.83        6.0  29.3  32.2   91.2
4390         24.7           8.2         2.72        6.9  30.1  33.2   90.8
4391         24.4           7.9         2.70        5.9  29.3  32.4   90.4
4392         26.6           8.7         2.94        5.9  29.6  32.7   90.5
4393         31.9          10.7         3.30        8.3  32.4  33.5   96.7
4394         26.3           8.2         4.18       10.9  19.6  31.2   62.9
4395         30.9          11.0         3.30        7.1  33.3  35.6   93.6
4396         28.2           9.5         2.86        6.1  33.2  33.7   98.6
4397         29.2          10.1         3.08       11.0  32.8  34.6   94.8
4398         29.3           9.8         3.26        8.2  30.1  33.4   89.9
4399         39.5          12.8         4.18        7.6  30.6  32.4   94.5
4400         23.2           7.7         2.55       10.2  30.2  33.2   91.0
4401         30.6          10.2         3.44        9.6  29.7  33.3   89.0
4402         29.1           9.5         3.24        8.9  29.3  32.6   89.8
4403         28.3           9.4         3.08        7.5  30.5  33.2   91.9
4404         31.9          10.3         3.42        8.7  30.1  32.3   93.3
4405         28.3           9.2         3.03        8.4  30.4  32.5   93.4
4406         33.7          10.4         3.48       11.4  29.9  30.9   96.8
4407         32.8          10.4         3.49        8.1  29.8  31.7   94.0
4408         33.7          10.8         3.67        6.7  29.4  32.0   91.8
4409         33.2          11.2         3.47        7.2  32.3  33.7   95.7
4410         31.5          10.4         3.15        9.1  33.0  33.0  100.0
4411         33.5          10.9         3.44        5.8  31.7  32.5   97.4

In [142]:
"""
Menampilkan 7 data teratas dari 11 kolom yang ada di dalam variabel
DataFrame "data_patient_treatment" tetapi dengan kondisi hanya data pasien rawat inap saja
"""
data_in = data_patient_treatment[data_patient_treatment["SOURCE"] == "in"]

data_in.head(7)

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
11         54.0          16.6         7.61       10.0           88  21.8   
12         31.7          10.4         4.91        9.7          348  21.2   
23         53.0          16.3         7.54        4.3          112  21.6   
24         34.6          11.2         4.42        7.0          152  25.3   
25         34.7          11.5         4.50        5.1          135  25.6   
26         32.5          10.6         4.19        1.6          111  25.3   
27         25.4           8.2         3.27        5.4           90  25.1   

    MCHC   MCV  AGE SEX SOURCE  
11  30.7  71.0    1   F     in  
12  32.8  64.6    1   M     in  
23  30.8  70.3    1   F     in  
24  32.4  78.3    1   F     in  
25  33.1  77.1    1   F     in  
26  32.6  77.6    1   F     in  
27  32.3  77.7    1   F     in

Output diatas menunjukkan bahwa variabel "data_in" menyimpan hasil penyaringan (filter) dengan kondisi yaitu menampilkan data dari kolom "SOURCE" yang bernilai in atau pasien yang berstatus sebagai rawat inap. Dengan kata lain nilai out (pasien berstatus sebagai rawat jalan) yang dimiliki oleh kolom "SOURCE" tidak tersimpan dan tidak ada di dalam variabel "data_in" karena variabel tersebut hanya khusus berisikan nilai in dari kolom "SOURCE" saja.

#### 1.1.1. Mencari trombosit rendah untuk orang dewasa normal (baik Pria/Wanita)

In [143]:
"""
Menampilkan 34 data terbawah dari keseluruhan kolom yang ada di dalam variabel DataFrame
"data_patient_treatment" dengan 3 kondisi yaitu pasien yang memiliki trombosit rendah untuk orang dewasa normal.
Trombosit rendah untuk orang dewasa normal menurut riset adalah kurang dari 150 (< 150)
dan range umur dari orang dewasa normal berkisar 19 - 65 tahun (>= 19 dan <= 65)

Ref: https://www.nhlbi.nih.gov/health/thrombocytopenia#:~:text=Thrombocytopenia%20is%20a%20condition%20that,is%20called%20a%20blood%20clot.

"""
low_thrombocyte = data_patient_treatment.loc[(data_patient_treatment["THROMBOCYTE"] < 150) &
                                             (data_patient_treatment["AGE"] >= 19) &
                                             (data_patient_treatment["AGE"] <= 65)]
low_thrombocyte.tail(34)

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
2920         38.6          13.6         4.54        7.3           50  30.0   
2950         29.3          10.2         3.63        5.0           94  28.1   
2970         37.1          12.4         4.65        4.6          130  26.7   
2979         25.3           8.7         2.87        4.8           85  30.3   
2984         23.5           8.0         2.76        4.0           60  29.0   
3005         31.4          10.4         3.96       17.9          148  26.3   
3049         40.9          14.3         4.67        2.3          128  30.6   
3060         37.7          13.2         4.30        2.0           77  30.7   
3084         40.7          14.1         4.59        4.3           35  30.7   
3085         40.6          14.0         4.56        3.8           17  30.7   
3086         40.0          13.6         4.49        4.4           15  30.3   
3087         39.6          13.6         4.42        4.7           36  30.8   
3088         37.6          12.8         4.22        5.0           40  30.3   
3090         36.5          12.5         4.08        5.0          116  30.6   
3171         48.4          16.0         5.67        3.0           60  28.2   
3203         45.1          15.5         4.68        8.4          120  33.1   
3234         31.9          10.4         3.78        4.4          135  27.5   
3292         24.5           8.3         2.50        5.3          121  33.2   
3315         30.8          10.9         3.45        9.2          121  31.6   
3316         27.0           9.3         2.98       14.4          147  31.2   
3317         30.2          10.1         3.28       26.8          129  30.8   
3318         25.9           8.7         2.81       15.6          121  31.0   
3329         38.9          12.5         4.53        5.1          129  27.6   
3333         21.9           8.4         2.39        8.8           32  35.1   
3336         26.7           9.9         2.92        8.7           30  33.9   
3349         28.0           9.5         3.36       14.1          114  28.3   
3350         25.2           8.5         3.01       12.5          105  28.2   
3356         25.1           9.7         2.77        7.4            8  34.7   
3358         26.4          10.1         2.97        7.1           25  34.0   
3362         34.6          11.7         4.16        3.1           85  28.1   
3375         28.8           9.7         3.46       12.1          128  28.0   
3376         27.2           9.0         3.18       11.3          145  28.3   
3385         26.9           9.3         3.13        5.7           20  29.7   
3388         24.9           9.7         2.73        4.8           15  35.5   

      MCHC   MCV  AGE SEX SOURCE  
2920  35.2  85.0   59   M     in  
2950  34.8  80.7   59   F     in  
2970  33.4  79.8   59   M     in  
2979  34.4  88.2   60   M    out  
2984  34.0  85.1   60   M     in  
3005  33.1  79.3   60   M     in  
3049  35.0  87.6   61   F    out  
3060  35.0  87.7   61   F     in  
3084  34.6  88.7   61   F     in  
3085  34.5  89.0   61   F     in  
3086  34.0  89.1   61   F     in  
3087  34.3  89.6   61   F     in  
3088  34.0  89.1   61   F     in  
3090  34.2  89.5   61   F     in  
3171  33.1  85.4   62   F    out  
3203  34.4  96.4   63   M    out  
3234  32.6  84.4   63   M    out  
3292  33.9  98.0   64   M    out  
3315  35.4  89.3   64   M     in  
3316  34.4  90.6   64   M     in  
3317  33.4  92.1   64   M     in  
3318  33.6  92.2   64   M     in  
3329  32.1  85.9   65   F    out  
3333  38.4  91.6   65   F    out  
3336  37.1  91.4   65   F    out  
3349  33.9  83.3   65   F     in  
3350  33.7  83.7   65   F     in  
3356  38.2  90.6   65   F    out  
3358  38.3  88.9   65   F    out  
3362  33.8  83.2   65   M    out  
3375  33.7  83.2   65   F     in  
3376  33.1  85.5   65   F     in  
3385  34.6  85.9   65   F    out  
3388  39.0  91.2   65   F    out

Hasil diatas menunjukkan bahwa proses penyaringan data menggunakan 3 kondisi sudah sangat tepat dan berhasil. Variabel "low_thrombocyte" hanya menyimpan data-data trombosit pasien dewasa yang rendah dan tentunya pasien dewasa yang memiliki trombosit rendah tersebut harus berada di rentang umur orang dewasa yaitu 19 - 65 tahun. Walaupun pasien tersebut berada di rentang umur orang dewasa normal (19 - 65 tahun) tetapi pasien tersebut memiliki trombosit yang tinggi atau lebih dari 150, maka pasien tersebut tidak memenuhi syarat untuk disimpan di dalam variabel "low_thrombocyte". Begitu juga saat pasien yang memiliki trombosit rendah (kurang dari 150) tetapi tidak berada di rentang umur 19 - 65 tahun, maka pasien tersebut tidak termasuk ke dalam variabel "low_thrombocyte".

Selanjutnya saya akan menampilkan jumlah keseluruhan dari pasien dewasa yang memiliki trombosit yang rendah yaitu dibawah 150 dari variabel "low_thrombocyte" berdasarkan jenis kelamin pasien tersebut.

In [144]:
"""
Menampilkan jumlah frekuensi data dari kolom "SEX" untuk mengetahui
seberapa banyak Pria/Wanita dewasa yang memiliki trombosit rendah (< 150)
"""
low_thrombocyte["SEX"].value_counts().sort_values(ascending = True)

F    179
M    206
Name: SEX, dtype: int64

Dari output diatas menunjukkan bahwa Wanita dewasa yang memiliki trombosit rendah tidak sebanyak Pria dewasa normal karena jumlah wanita yang memiliki trombosit rendah hanya 179 orang sedangkan pasien dewasa berjenis kelamin Pria ternyata sangat banyak yang memiliki trombosit rendah yang kurang dari 150 dengan jumlah sebanyak 206 orang.

Selain itu, berdasarkan output diatas juga kita dapat mengetahui bahwa pasien yang memiliki trombosit rendah (< 150) dengan rentang umur 19 - 65 tahun (range umur dari orang dewasa) ternyata tidak terlalu banyak karena hanya berjumlah 385 orang saja. Hal ini belum dapat dipastikan secara pasti faktor yang mempengaruhi trombosit rendah di pasien dewasa dan dibutuhkan analisis lebih lanjut.

Selanjutnya saya akan mencari tau terkait umur dari Pria & Dewasa normal yang paling banyak/memiliki kadar trombosit rendah yaitu kurang dari 150.

In [145]:
"""
Menampilkan jumlah frekuensi data dari kolom "AGE" untuk mengetahui
seberapa banyak umur Pria & Wanita Dewasa yang memiliki trombosit rendah (< 150)
"""
low_thrombocyte["AGE"].value_counts().sort_values(ascending = False)

45    37
25    26
22    16
26    15
37    13
29    13
47    12
34    12
30    12
65    12
46    11
36    11
44    11
32    10
38    10
23    10
48     9
53     9
24     9
35     8
55     8
61     8
58     7
59     7
57     7
39     6
51     6
56     5
49     5
64     5
21     5
41     5
40     5
33     5
54     5
50     4
43     4
19     4
52     4
20     3
60     3
31     3
63     2
42     1
62     1
27     1
Name: AGE, dtype: int64

Berdasarkan hasil output diatas dapat diketahui jika pasien dewasa (baik pasien berjenis kelamin Pria/Wanita) yang berumur 45 tahun paling sering/paling banyak mengalami penurunan kadar trombosit di dalam sel darah mereka. Hal ini sangat berisiko sebab jika orang dewasa normal memiliki kadar trombosit yang rendah, baik Wanita/Pria, akan sangat rentan mengalami pendarahan yang tidak terkontrol dan alhasil tidak dapat sistem pembekuan darah tidak dapat bekerja dengan sangat baik dalam menyembuhkan luka/menghentikan pendarahan.

Analisis selanjutnya yaitu saya ingin menampilkan informasi terkait ada berapa banyak pasien yang memiliki kadar eritrosit yang terendah ataupun tertinggi berdasarkan jenis kelamin pasien atau dengan kata lain saya ingin mengetahui berapa banyak pasien yang terindikasi memiliki kelebihan/kekurangan sel darah merah (eritrosit).

#### 1.1.2. Menampilkan jumlah kadar eritrosit terendah dan tertinggi berdasarkan jenis kelamin pasien

In [146]:
"""
Melakukan proses penyaringan berdasarkan kolom data (filter)
dengan hanya menampilkan data-data dari kolom "SEX" dan "ERYTHROCYTE"
"""
data_eritrosit = data_patient_treatment.filter(items=["SEX", "ERYTHROCYTE"])
data_eritrosit.tail(12)

SEX  ERYTHROCYTE
4400   F         2.55
4401   F         3.44
4402   F         3.24
4403   F         3.08
4404   F         3.42
4405   F         3.03
4406   F         3.48
4407   F         3.49
4408   F         3.67
4409   F         3.47
4410   F         3.15
4411   F         3.44

In [147]:
""""
Melakukan penyaringan data kembali dari variabel "data_eritrosit" dengan kondisi kadar
eritrosit harus kurang dari 4.60 (< 4.60) untuk pasien berjenis kelamin Pria dan Wanita
"""
low_erythrocyte_male_female = data_eritrosit[data_eritrosit["ERYTHROCYTE"] < 4.60]

low_erythrocyte_male_female.head()

SEX  ERYTHROCYTE
4    M         4.23
5    M         4.53
8    F         4.54
10   M         4.57
13   M         4.40

In [148]:
# Menampilkan jumlah pasien berdasarkan jenis kelamin yang memiliki kandungan sel darah merah (eritrosit) terendah
low_erythrocyte_male_female["SEX"].value_counts().sort_values(ascending = True)

M     878
F    1391
Name: SEX, dtype: int64

Berdasarkan hasil diatas diketahui bahwa pasien dengan jenis kelamin Perempuan (ditulis dengan nilai F yaitu Female) paling banyak mengalami kekurangan eritrosit atau sel darah merah dibandingkan Pasien berjenis kelamin Pria. Hal ini dibuktikan setelah variabel "low_erythrocyte_male_female" menyimpan hasil filtering data dari kolom "ERYTHROCYTE" yang dimiliki oleh variabel "data_eritrosit" dimana variabel tersebut (variabel "data_eritrosit") hanya memiliki 2 kolom saja yaitu "SEX" dan "ERYTHROCYTE".

Jumlah pasien perempuan yang mengalami kondisi rendahnya kadar eritrosit kurang dari 4.60 ada sebanyak 1.391 pasien perempuan sedangkan pasien pria hanya terdapat 878 orang yang mengalami rendahnya kadar eritrosit. Selanjutnya saya akan mencari berapa banyak pasien yang mengalami tingginya kadar eritrosit atau dengan kata lain memiliki lebih banyak kandungan sel darah merah di dalam tubuhnya.

In [149]:
""""
Memfilter data dari variabel "data_eritrosit" dengan kondisi kadar
eritrosit lebih dari 5.70 (> 5.70) untuk pasien berjenis kelamin Pria dan Wanita
"""
high_erythrocyte_male_female = data_eritrosit[data_eritrosit["ERYTHROCYTE"] > 5.70]

high_erythrocyte_male_female.tail(9)

SEX  ERYTHROCYTE
3643   M         7.30
3695   F         6.28
3696   F         6.64
3795   M         5.77
3803   F         6.09
3828   M         5.98
3950   M         6.11
4152   F         5.93
4267   M         6.00

In [150]:
# Menampilkan jumlah pasien berdasarkan jenis kelamin yang memiliki kandungan sel darah merah (eritrosit) tertinggi
high_erythrocyte_male_female["SEX"].value_counts().sort_values(ascending = False)

M    210
F     36
Name: SEX, dtype: int64

Hasil diatas menunjukkan bahwa sebanyak 210 pasien berjenis kelamin Pria/Laki-Laki memiliki kandungan sel darah merah diatas atau melebihi 5.70 juta/mikroliter darah sedangkan hanya 36 pasien Perempuan yang memiliki kandungan sel darah merah melebihi 5.70 juta/mikroliter. Harus diingat bahwa memiliki kandungan sel darah merah yang melebihi batas normal sangat berbahaya sebab menurut riset dapat mengindikasikan tubuh kita mengalami dehidrasi, kanker/tumor yang muncul pada bagian tubuh tertentu, adanya kelainan pada hemoglobin, efek samping dari konsumsi obat, gangguan tidur, dll.

Analisis terakhir dengan menggunakan teknik filter adalah mencari tau jumlah leukosit/kandungan sel darah putih untuk orang dewasa yang berada di range usia 19 - 65 tahun berdasarkan status rawat pasien (apakah pasien tersebut dikategorikan sebagai pasien rawat inap/rawat jalan) dan juga berdasarkan jenis kelamin pasien.

#### 1.1.3. Menampilkan jumlah total & rata-rata kadar dari kandungan leukosit untuk setiap pasien dewasa

In [151]:
"""
Melakukan proses penyaringan berdasarkan kolom data (filter)
dengan hanya menampilkan data-data dari kolom "AGE" dan "LEUCOCYTE"
"""
data_leukosit = data_patient_treatment.filter(items=["AGE", "LEUCOCYTE"])

data_leukosit.head(2)

AGE  LEUCOCYTE
0    1        6.3
1    1       12.7

Setelah menyaring (filtering) data terhadap kolom "AGE" dan "LEUCOCYTE" dari variabel DataFrame "data_patient_treatment" dan disimpan di dalam variabel baru yaitu "data_leukosit", tahap selanjutnya adalah menyaring data berdasarkan rentang umur pasien dewasa (19 - 65 tahun). Hal ini dilakukan agar analisis sederhana yang dilakukan yaitu untuk melihat jumlah total keseluruhan ataupun nilai rata-rata dari kandungan leukosit yang dimiliki oleh para pasien dewasa dapat terfokus dan pembahasan dari analisisnya tidak meluas ke lain topik.

In [152]:
"""
Memfilter data yang ada di dalam kolom "AGE" dengan memprioritaskan umur pasien dewasa normal (19 - 65 tahun)
untuk diketahui jumlah total dari kadar kandungan leokosit dan tampilkan 22 data teratas dari variabel baru
"""
filtering_age_for_adult_patient = data_leukosit.loc[((data_leukosit["AGE"] >= 19) & (data_leukosit["AGE"] <= 65))]

filtering_age_for_adult_patient.head(22)

AGE  LEUCOCYTE
476   19        6.4
477   19        5.6
478   19        8.0
479   19        5.8
480   19        9.1
481   19        5.8
482   19        7.4
483   19        3.4
484   19        9.2
485   19        6.0
486   19        2.5
487   19        7.0
488   19        2.0
489   19        4.0
490   19        4.3
491   19        3.4
492   19        4.0
493   19        1.5
494   19        5.2
495   19        3.1
496   19        9.1
497   19        4.6

In [153]:
"""
Menampilkan jumlah total keseluruhan (menggunakan fungsi "sum()") dari kadar kandungan leukosit/sel darah putih
yang dimiliki oleh para pasien dewasa. Dengan kata lain, kita akan menampilkan jumlah total keseluruhan dari kadar
kandungan sel darah putih tertinggi hingga terendah yang dialami oleh para pasien dewasa rumah sakit
"""
data_sum_leucocyte = filtering_age_for_adult_patient.groupby("AGE")["LEUCOCYTE"].sum().sort_values(ascending = False)

print(data_sum_leucocyte)

AGE
59    924.9
62    883.9
65    747.7
55    720.6
53    719.6
60    699.1
63    686.6
56    668.4
64    613.8
26    613.5
30    599.6
45    595.6
46    592.3
58    589.8
57    589.4
25    570.1
61    567.9
24    564.7
29    564.1
32    563.3
54    562.2
49    555.6
51    543.9
33    517.3
47    512.0
36    506.2
35    501.7
52    487.3
27    484.9
43    484.0
37    482.1
41    471.1
44    467.3
38    447.2
28    446.0
50    444.6
40    435.2
31    416.7
34    415.7
22    384.7
42    370.4
21    367.5
48    356.5
23    328.9
39    311.6
20    181.5
19    154.9
Name: LEUCOCYTE, dtype: float64


Hasil diatas menunjukkan bahwasanya pasien dewasa dengan umur 59 tahun, jika dihitung total secara keseluruhan berdasarkan hasil uji leukosi antar masing-masing pasien yang memiliki kisaran umur orang dewasa (19 - 65 tahun), maka pasien yang berumur 59 tahun memiliki jumlah total yang sangat tinggi dan terbanyak dalam hal memiliki sel darah putih (leukosit) yakni 924.9. Tetapi hal ini tidak berakibat buruk ke pasien karena pada dasarnya perhitungan diatas bukan untuk mengetahui SEBERAPA BANYAKNYA pasien berumur dewasa yang mengalami kadar leukosit terbanyak melainkan hanya untuk mengetahui JUMLAH TOTAL DARI KADAR KANDUNGAN LEUKOSIT per masing-masing rentang umur dewasa saja.

Selain informasi mengenai pasien yang berumur 59 tahun memiliki jumlah total dari kadar leukosit tertinggi (924.4), pasien dengan umur 19 tahun memiliki jumlah total terhadap kadar leukosit yang paling rendah diantara rentang umur yang lainnya yakni hanya 154.9. Hanya memiliki selisih nilai 26.6 dengan pasien yang berumur 20 tahun karena pasien dewasa yang masih berumur 20 tahun memiliki jumlah total dari kadar leukosit yang hanya sebesar 181.5 (terendah kedua setelah pasien berumue 19 tahun).

Tahap selanjutnya adalah melakukan proses mencari sekaligus menampilkan nilai rata-rata dari kolom "LEUCOCYTE" per masing-masing rentang umur pasien dewasa yang melakukan test pemeriksaan darah di dalam laboratorium.

In [154]:
"""
Menampilkan nilai rata-rata dari kadar kandungan leukosit/sel darah putih yang dimiliki oleh para pasien yang berada
di kisaran umur orang dewasa normal yaitu 19 - 65 tahun. Dengan kata lain, kita akan menampilkan rata-rata kadar
kandungan sel darah putih dari yang terendah yang dialami oleh para pasien dewasa rumah sakit
"""
mean_leucocyte = filtering_age_for_adult_patient.groupby("AGE")["LEUCOCYTE"].mean().sort_values(ascending = True)

print(mean_leucocyte)

AGE
19     5.341379
25     6.196739
39     6.629787
23     6.712245
48     6.726415
24     7.058750
45     7.090476
22     7.124074
33     7.184722
38     7.331148
29     7.422368
30     7.495000
34     7.558182
20     7.562500
47     7.641791
37     7.775806
44     7.788333
32     7.823611
46     7.897333
36     7.909375
26     7.967532
21     7.989130
31     8.170588
28     8.259259
52     8.259322
57     8.420000
61     8.604545
42     8.613953
40     8.704000
41     8.724074
43     8.962963
27     8.979630
58     9.215625
55     9.238462
35     9.290741
49     9.416949
50     9.459574
63     9.670423
56     9.686957
54     9.693103
51     9.712500
64     9.742857
60     9.987143
65    10.242466
62    10.398824
53    10.903030
59    11.707595
Name: LEUCOCYTE, dtype: float64


Berdasarkan hasil output diatas mengenai nilai rata-rata dari kandungan sel darah putih yang dmiliki oleh berbagai macam range umur pasien dewasa dapat diketahui bahwasanya pasien berumur 19 tahun memiliki rata-rata dari jumlah kandungan leukosit hanya sebesar 5.341379. Nilai rata-rata tersebut menjadikan pasien berumur 19 tahun merupakan pasien dewasa yang memiliki rata-rata kadar dari jumlah leukosit yang sangat rendah dan tidak banyak. Tetapi hal tersebut tidak terlalu aneh mengingat saat kita menampilkan jumlah total dari keseluruhan kadar kandungan leukosit, pasien yang berumur 19 tahun memiliki nilai total kandungan leukosit yang jauh sangat rendah yaitu hanya 154.9.

Disamping itu, pasien yang berumur 59 tahun memiliki rata-rata kandungan leukosit yang sangat tinggi karena pasien dengan umur 59 tahun menghasilkan rata-rata kadar kandungan leukosit dengan jumlah sebanyak 11.707595.

## 2. Data Preprocessing

Setelah melakukan proses Exploratory Data Analysis menggunakan beberapa fungsi di dalam library Pandas serta juga melakukan analisis data sederhana, pada tahapan kali ini akan dilakukan beberapa proses pembersihan data karena untuk saat ini susunan data yang ada di dalam variabel DataFrame "data_patient_treatment" masih bersifat data kotor (dirty data). Maka dari itu untuk mengatasi permasalahan yang terjadi dengan dirty data yang saat ini berada di dalam variabel DataFrame "data_patient_treatment", saya akan melakukan beberapa proses pembersihan data yaitu sebagai berikut:

1. Melakukan perubahan format dalam penamaan kolom data yang semula ditulis dengan huruf kapital akan ditransformasikan menjadi huruf berformat str.title yaitu setiap huruf pertama di awal kalimat adalah kapital (H) dan huruf selanjutnya diikuti dengan huruf kecil (e).
2. Mengubah nilai sekaligus tipe data yang ada di dalam kolom "Sex" dan "Status" menjadi bilangan binary dengan fungsi LabelEncoder dari Sklern.preprocessing.
3. Mengecek kembali nilai data duplikat yang ada di dalam variabel DataFrame "data_patient_treatment". Jika terdapat data duplikat maka akan segera dihilangkan dari baris data.
4. Mengecek kembali missing value/nilai data yang hilang yang ada di dalam masing-masing kolom dari variabel DataFrame "data_patient_treatment". Jika terdapat missing value di dalam suatu kolom maka akan segera dihapus dari baris data di kolom tersebut supaya data menjadi lebih konsisten dan terhindar dari mis-interpretasi.
5. 

In [155]:
# Mengubah format penamaan kolom yang ada di dalam variabel DataFrame "data_patient_treatment"
data_patient_treatment = data_patient_treatment.rename(columns={"HAEMATOCRIT": "Hematocrit", "HAEMOGLOBINS": "Hemoglobin",
                                                                "ERYTHROCYTE": "Erythrocyte", "LEUCOCYTE": "Leucocyte",
                                                                "THROMBOCYTE": "Thrombocyte", "MCH": "Mch", "MCHC": "Mchc",
                                                                "MCV": "Mcv", "AGE": "Age", "SEX": "Sex", "SOURCE": "Status"})

data_patient_treatment.columns

Index(['Hematocrit', 'Hemoglobin', 'Erythrocyte', 'Leucocyte', 'Thrombocyte',
       'Mch', 'Mchc', 'Mcv', 'Age', 'Sex', 'Status'],
      dtype='object')

In [156]:
"""
Melakukan transformasi/perubahan nilai dan tipe data untuk kolom "Sex" dan "Status" dari yang semula
string kategorikal menjadi bilangan integer numerik serta menampilkan 38 data teratas
"""
label_encoding_for_changing_data = LabelEncoder()

data_patient_treatment["Sex"] = label_encoding_for_changing_data.fit_transform(data_patient_treatment["Sex"])
data_patient_treatment["Status"] = label_encoding_for_changing_data.fit_transform(data_patient_treatment["Status"])

data_patient_treatment.head(38)

Hematocrit  Hemoglobin  Erythrocyte  Leucocyte  Thrombocyte   Mch  Mchc  \
0         35.1        11.8         4.65        6.3          310  25.4  33.6   
1         43.5        14.8         5.39       12.7          334  27.5  34.0   
2         33.5        11.3         4.74       13.2          305  23.8  33.7   
3         39.1        13.7         4.98       10.5          366  27.5  35.0   
4         30.9         9.9         4.23       22.1          333  23.4  32.0   
5         34.3        11.6         4.53        6.6          185  25.6  33.8   
6         31.1         8.7         5.06       11.1          416  17.2  28.0   
7         40.3        13.3         4.73        8.1          257  28.1  33.0   
8         33.6        11.5         4.54       11.4          262  25.3  34.2   
9         35.4        11.4         4.80        2.6          183  23.8  32.2   
10        33.7        11.5         4.57       13.2          322  25.2  34.1   
11        54.0        16.6         7.61       10.0           88  21.8  30.7   
12        31.7        10.4         4.91        9.7          348  21.2  32.8   
13        35.3        11.9         4.40        5.8          205  27.0  33.7   
14        34.5         9.8         5.75       15.4          548  17.0  28.4   
15        34.0        10.3         5.27       16.2          572  19.5  30.3   
16        35.0        11.6         4.58        7.4          154  25.3  33.1   
17        51.3        15.7         7.24        4.8          129  21.7  30.6   
18        31.3        10.8         4.02        7.9          250  26.9  34.5   
19        36.8        12.9         4.67        5.7          235  27.6  35.1   
20        34.9        11.6         4.71        9.5          275  24.6  33.2   
21        36.5        11.9         4.74       10.0          394  25.1  32.6   
22        33.7        11.0         4.69       18.4          260  23.5  32.6   
23        53.0        16.3         7.54        4.3          112  21.6  30.8   
24        34.6        11.2         4.42        7.0          152  25.3  32.4   
25        34.7        11.5         4.50        5.1          135  25.6  33.1   
26        32.5        10.6         4.19        1.6          111  25.3  32.6   
27        25.4         8.2         3.27        5.4           90  25.1  32.3   
28        36.8        11.9         4.69        1.9          108  25.4  32.3   
29        35.4        11.5         4.45        5.7           85  25.8  32.5   
30        36.1        11.7         4.51        6.4           85  25.9  32.4   
31        37.3        11.6         4.60        4.1          115  25.2  31.1   
32        44.2        14.3         5.41       14.5          125  26.4  32.4   
33        41.2        13.0         4.88       15.0          178  26.6  31.6   
34        41.0        12.8         4.76       13.8          192  26.9  31.2   
35        43.6        13.2         4.90       14.1          212  26.9  30.3   
36        38.4        12.7         4.68        3.4          136  27.1  33.1   
37        33.4        11.0         4.29       14.9          275  25.6  32.9   

     Mcv  Age  Sex  Status  
0   75.5    1    0       1  
1   80.7    1    0       1  
2   70.7    1    0       1  
3   78.5    1    0       1  
4   73.0    1    1       1  
5   75.7    1    1       1  
6   61.5    1    0       1  
7   85.2    1    0       1  
8   74.0    1    0       1  
9   73.8    1    0       1  
10  73.7    1    1       1  
11  71.0    1    0       0  
12  64.6    1    1       0  
13  80.2    1    1       1  
14  60.0    1    1       1  
15  64.5    1    1       1  
16  76.4    1    0       1  
17  70.9    1    0       1  
18  77.9    1    0       1  
19  78.8    1    0       1  
20  74.1    1    0       1  
21  77.0    1    0       1  
22  71.9    1    0       1  
23  70.3    1    0       0  
24  78.3    1    0       0  
25  77.1    1    0       0  
26  77.6    1    0       0  
27  77.7    1    0       0  
28  78.5    1    0       0  
29  79.6    1    0       0  
30  80.0    1    0     

In [157]:
# Menampilkan tipe data dari 11 kolom yang ada di dalam "data_patient_treatment" setelah dilakukan proses transformasi
data_patient_treatment.dtypes

Hematocrit     float64
Hemoglobin     float64
Erythrocyte    float64
Leucocyte      float64
Thrombocyte      int64
Mch            float64
Mchc           float64
Mcv            float64
Age              int64
Sex              int32
Status           int32
dtype: object

Dengan menggunakan fungsi LabelEncoder dari library SKlearn.preprocessing, saya telah berhasil mentransformasikan nilai dan tipe data dari kolom "Sex" dan "Status" menjadi data binary numerik (0 dan 1) dengan tipe data integer (int). Karena sebelumnya kedua kolom tersebut menggunakan data string kategorik yang masing-masing kedua kolom tersebut memiliki 2 jenis nilai data yaitu "F" dan "M" (Female dan Male) dari kolom "Sex" serta "out" dan "in" (status out untuk pasien rawat jalan dan status in untuk pasien rawat inap) dari kolom "Status".

Alasan saya melakukan proses perubahan nilai dan tipe data dari kolom "Sex" dan "Status" adalah karena pada saat melakukan proses klasifikasi dengan Algoritma Machine Learning atau melakukan proses standarisasi nilai data, di dalam variabel DataFrame tidak boleh memiliki data string atau kolom yang masih bertipe data object. Maka dari itu untuk mengatasi permasalahan dan menghindari error di langkah ke depannya, saya melakukan proses perubahan nilai data dari kategorikal menjadi bilangan numerikal, khususnya untuk data yang ada di dalam kolom "Sex" dan "Status".

In [158]:
# Mengecek kembali apakah di dalam variabel DataFrame "data_patient_treatment" terdapat data duplikat/tidak
data_patient_treatment.duplicated().sum()

0

In [159]:
"""
Setelah mengecek kembali apakah di dalam "data_patient_treatment" terdapat data duplikat/tidak,
selanjutnya adalah melakukan proses pengecekan kembali apakah masing-masing
kolom yang ada di dalam "data_patient_treatment" terdapat missing value/tidak
"""

data_patient_treatment.isnull().sum()

Hematocrit     0
Hemoglobin     0
Erythrocyte    0
Leucocyte      0
Thrombocyte    0
Mch            0
Mchc           0
Mcv            0
Age            0
Sex            0
Status         0
dtype: int64

In [160]:
"""
Menampilkan nilai data unik yang dimiliki oleh kolom "Age" serta
menjumlahkan berapa banyak data unik yang dimiliki oleh kolom "Age"
"""
print(data_patient_treatment["Age"].unique(), "\n")

print("Jumlah Data Unik dari Kolom Age adalah", data_patient_treatment["Age"].nunique(), "data")

[ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48
 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72
 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 98 99] 

Jumlah Data Unik dari Kolom Age adalah 95 data


Berdasarkan hasil diatas diketahui terdapat 95 rentang umur pasien yang berbeda-beda. Dimulai dari umur 1 - 93 tahun dan 98 - 99 tahun. Walaupun umur pasien tertinggi adalah 99 tahun tetapi pada kenyataannya jumlah data unik yang ada di dalam kolom "Age" hanya terdapat 95 data saja sebab nilai 94 - 97 tidak ada di dalam kolom "Age".

Tetapi yang ingin saya permasalahkan adalah bukan hal tersebut melainkan nilai yang ada di dalam kolom "Age" seakan-akan dibuat secara berurutan dan tidak berdasarkan logika studi kasus. Seperti sebagai contoh pasien yang memiliki umur 1 tahun hampir sangat tidak mungkin untuk dilakukan uji laboratorium. Maka dari itu pada eksperimen kali ini, saya ingin memisahkan data pasien yang memiliki kisaran/rentang umur 1 - 16 tahun dan 68 - 99 tahun karena pada rentang umur tersebut, pasien masih sangat kecil maupun sudah terlalu tua.

Data dari hasil pemisahan rentang umur tersebut akan disimpan di dalam variabel DataFrame baru yang diberi nama "new_df_patient". Hal tersebut dilakukan agar tidak menganggu keaslian data yang dimiliki oleh variabel DataFrame yang asli yaitu "data_patient_treatment".

In [161]:
"""
Meng-copy variabel DataFrame "data_patient_treatment" dan menyalin semua data yang ada di dalamnya
serta disimpan di dalam variabel DataFrame baru yaitu "new_df_patient". Hal ini dilakukan
agar keasilan data di dalam "data_patient_treatment" dapat terjaga dan tidak terutak-atik dengan beberapa
percobaan yang akan dilakukan
"""
new_df_patient = data_patient_treatment.copy()

In [162]:
# Menampilkan 5 data teratas yang dimiliki oleh variabel DataFrame asli (data_patient_treatment) dan copy (new_df_patient)
print("--------------------------------------DataFrame Asli--------------------------------------")
print(data_patient_treatment.head(), "\n")

print("--------------------------------------DataFrame Copy--------------------------------------")
print(new_df_patient.head())

--------------------------------------DataFrame Asli--------------------------------------
   Hematocrit  Hemoglobin  Erythrocyte  Leucocyte  Thrombocyte   Mch  Mchc  \
0        35.1        11.8         4.65        6.3          310  25.4  33.6   
1        43.5        14.8         5.39       12.7          334  27.5  34.0   
2        33.5        11.3         4.74       13.2          305  23.8  33.7   
3        39.1        13.7         4.98       10.5          366  27.5  35.0   
4        30.9         9.9         4.23       22.1          333  23.4  32.0   

    Mcv  Age  Sex  Status  
0  75.5    1    0       1  
1  80.7    1    0       1  
2  70.7    1    0       1  
3  78.5    1    0       1  
4  73.0    1    1       1   

--------------------------------------DataFrame Copy--------------------------------------
   Hematocrit  Hemoglobin  Erythrocyte  Leucocyte  Thrombocyte   Mch  Mchc  \
0        35.1        11.8         4.65        6.3          310  25.4  33.6   
1        43.5        14

In [165]:
"""
Memfilter data yang ada di kolom "Age" dengan menghilangkan nilai umur 1 - 16 tahun dan 68 - 99 tahun dan hasil filter
tersebut disimpan di dalam variabel DataFrame copy yaitu "new_df_patient". Itu berarti setiap ada pasien yang memiliki
kisaran umur 1 - 16 tahun/68 - 99 tahun sudah tidak ada di dalam "new_df_patient"
"""
new_df_patient = new_df_patient.loc[((new_df_patient["Age"] >= 17) & (new_df_patient["Age"] <= 67))]

new_df_patient.tail(37)

Hematocrit  Hemoglobin  Erythrocyte  Leucocyte  Thrombocyte   Mch  Mchc  \
3534        37.7        12.8         4.41       14.4          284  29.0  34.0   
3535        39.1        13.1         4.49       10.2          540  29.2  33.5   
3536        36.5        11.9         4.20        9.7          333  28.3  32.6   
3537        36.2        12.3         4.38       11.6          332  28.1  34.0   
3538        38.9        12.9         4.03        5.8          191  32.0  33.2   
3539        37.6        12.7         4.52       22.5          421  28.1  33.8   
3540        25.9         9.0         2.90       10.6           85  31.0  34.7   
3541        34.4        11.0         3.89       26.0          386  28.3  32.0   
3542        32.4        10.3         4.15       11.3          481  24.8  31.8   
3543        31.7        10.1         3.62       21.9          353  27.9  31.9   
3544        35.5        11.4         4.52        9.5          406  25.2  32.1   
3545        23.6         8.0         3.39       24.4          334  23.6  33.9   
3546        34.6        10.8         3.86       21.1          346  28.0  31.2   
3547        34.6        10.6         3.82       19.0          308  27.7  30.6   
3548        29.8         9.8         4.03       16.2          263  24.3  32.9   
3549        33.6        10.3         3.72       14.0          283  27.7  30.7   
3550        38.5        12.2         4.84        9.6          443  25.2  31.7   
3551        36.5        12.4         4.21        9.1          294  29.5  34.0   
3552        35.5        12.2         4.25       18.7          292  28.7  34.4   
3553        45.5        14.7         5.42        8.7          317  27.1  32.3   
3554        46.5        15.8         5.07        7.0          267  31.2  34.0   
3555        43.0        14.5         4.45        6.2          255  32.6  33.7   
3556        47.4        15.7         5.71        7.2          231  27.5  33.1   
3557        40.3        13.5         4.53        7.6          254  29.8  33.5   
3558        40.4        13.4         4.59        6.7          278  29.2  33.2   
3559        42.7        14.0         5.15       11.0          338  27.2  32.8   
3560        36.5        11.9         4.43        7.9          246  26.9  32.6   
3561        25.1         8.1         2.66        9.7          131  30.5  32.3   
3562        31.6        10.4         3.86       10.9          242  26.9  32.9   
3563        40.8        14.4         3.53        5.3          309  40.8  35.3   
3564        40.9        13.4         4.49        5.3          190  29.8  32.8   
3565        31.3        10.3         4.23       15.8          278  24.3  32.9   
3566        31.7        10.6         3.62        7.9          349  29.3  33.4   
3567        32.5        10.9         3.93       17.8          198  27.7  33.5   
3568        34.5        11.3         3.73        6.9          207  30.3  32.8   
3569        35.8        12.3         4.45       10.4          495  27.6  34.4   
3570        36.6        12.3         4.47       10.8          572  27.5  33.6   

        Mcv  Age  Sex  Status  
3534   85.5   67    1       0  
3535   87.1   67    1       0  
3536   86.9   67    1       0  
3537   82.6   67    1       0  
3538   96.5   67    1       0  
3539   83.2   67    0       0  
3540   89.3   67    0       0  
3541   88.4   67    0       0  
3542   78.1   67    0       0  
3543   87.6   67    0       0  
3544   78.5   67    0       0  
3545   69.6   67    0       0  
3546   89.6   67    0       0  
3547   90.6   67    0       0  
3548   73.9   67    0       0  
3549   90.3   67    0       0  
3550   79.5   67    0       0  
3551   86.7   67    1       1  
3552   83.5   67    1       1  
3553   83.9   67    1       1  
3554   91.7   67    1       1  
3555   96.6   67    1       1  
3556   83.0   67    1       1  
3557   89.0   67    1       1  
3558   88.0   67    0       1  
3559   82.9   67    0       1  
3560   82.4   67    0       1  
3561   94.4   67    0       1  

In [167]:
len(data_patient_treatment)

4412

In [166]:
# Menampilkan jumlah data yang saat ini dimiliki oleh variabel DataFrame "new_df_patient"
len(new_df_patient)

3172